docs/adr/0003-graphql-layer-between-backend-frontend.md

```diff
@@ -0,0 +1,104 @@
+# GraphQL layer between backend and frontend
+
+Date: 2020-04-30
+
+## Related tickets (not available for public)
+
+SIG-2568
+
+## Disclaimer
+
+Since this proposal is quite big the original author (jpoppe) has taken the liberty make this
+proposal a bit more opiniated then it normally should be. (One of the reasons is that the original
+author has done this implementation and fine tuned it a couple of times)
+
+## Context
+
+Since our current backend only serves data through REST and there is no possibility to generate
+generate types/interfaces there is no automated way to validate data on the frontend which is coming
+from the backend.
+
+Besides of that there are too many inconstencies between different backend calls and the current
+backend API documentation is not always correct and/or up to date because it's written manually
+and not generated from code.
+
+Because of this our `React components` are over complicated since we need to mangle data all the
+time to be able to send a POST/PUT request from data which got from a GET request.
+
+## Proposal
+
+We would like to introduce a thin layer between the backend and our frontend which should expose
+backend data through GraphQL Queries and should expose backend PUT/POST/DELETE actions via GraphQL
+mutations.
+
+We will also need to implement DTO's (Data Transfer Objects) so we can create/use data contracts
+between backend and frontend.
+
+## How
+
+Since this proposal will take quite some time/impact to implement I propose to start a POC with one
+endpoint which we could make acccessible through GraphQL.
+
```

Comment:
I would include a section 'Prerequisites' in which we describe what needs to be in place (for instance, the mono repo), before we can set up this infrastructure.

Reply:
No list of related technical implementations (mono-repo et al.) that need to be in place?

Reply:
Yes we discussed that, did not find time yet, will do this tomorrow ;)